cache/cache_data.go

```diff
@@ -17,7 +17,7 @@ type cache struct {
 	// When the item is accessed, it's moved to the end of the list.
 	usage listItem
 
-	lock sync.Mutex
+	lock sync.RWMutex
```

Comment:
First of all, it's not always the right choice to just use RWMutex everywhere - we should see the real numbers to ensure this is better.  Can you prove why it is better here by doing some kind of a benchmark?

Reply:
I will try to run some concurrent workloads with both version.
I might be able to compare them tomorrow.

Could you take a look at 00e2bceb0837b7e0510f9eb9bd3ffc9464cbea73 in the mean time?